pkg/envoy/lds/inmesh.go

```diff
@@ -27,36 +26,78 @@ var (
 	supportedDownstreamHTTPProtocols = []string{"http/1.0", "http/1.1", "h2c"}
 )
 
-func getInboundInMeshFilterChain(proxyServiceName service.MeshService, cfg configurator.Configurator) (*xds_listener.FilterChain, error) {
-	marshalledDownstreamTLSContext, err := ptypes.MarshalAny(envoy.GetDownstreamTLSContext(proxyServiceName, true /* mTLS */))
+func (lb *listenerBuilder) getInboundMeshFilterChains(proxyService service.MeshService) []*xds_listener.FilterChain {
+	var filterChains []*xds_listener.FilterChain
+
+	// Apply an inbound HTTP filter chain to filter in-mesh traffic
+	if httpFilterChain, err := lb.getInboundMeshHTTPFilterChain(proxyService); err != nil {
+		log.Error().Err(err).Msgf("Error constructing inbound mesh filter chain for proxy %s", proxyService)
+	} else {
+		filterChains = append(filterChains, httpFilterChain)
+	}
+
+	// TODO: add TCP filter chain here
+
+	return filterChains
+}
+
+func (lb *listenerBuilder) getInboundHTTPFilters(proxyService service.MeshService) ([]*xds_listener.Filter, error) {
```

Comment:
Do you see this function ever taking a proxy identity instead of a service?

Reply:
Not at this moment, because the inbound filters make use of attributes such as SNI that are configured per service.